docs/general/dql/selects.rst

```diff
@@ -736,8 +736,8 @@ with ``landmarks[n]``, where ``n`` is the integer array index, like so::
 
 .. NOTE::
 
-    The first index value is ``1``. The maximum array index is ``2147483648``.
-    Using an index greater than the array size results in a NULL value.
+    The minimum index value is ``-2147483648``. The maximum array index is
+    ``2147483647``. Using an index out of the range results in an exception.
 
```

Comment:
> Let's add a note that if the index is < 1 then NULL is returned,

[Updating single array element](https://github.com/crate/crate/issues/13649) should allow something like `update t set a[-1] = 1`.

> but please check also what happens now that we remove the < 1 from SubscriptValidator.

`SubscriptValidator` cannot validate negative expressions since it is not evaluated yet(For example, `-1` is an `IntegerLiteral` with value `1` wrapped in `NegativeExpression`). 

Btw, I have a test in `SubscriptFunctionTest`:

```
    @Test
     public void test_valid_min_and_max_array_index_access() {
         assertNormalize("subscript([], 2147483647)", isLiteral(null));
         assertNormalize("subscript([], -2147483648)", isLiteral(null));
     }
```

which goes through `SubscriptValidator` as well.

Reply:
Are we going to allow 0 and negative indices as postgres? I don't think so  (/cc @mfussenegger)